### PR TITLE
OrphanTypeDeclaration Error explaination

### PR DESCRIPTION
**Pull request by antarestrader**
Added basic explanation for this error.

### CODE REVIEW

File: errors/OrphanTypeDeclaration.md [hunk 1]
```diff
@@ -5,17 +5,23 @@
...
+addToo a b = a + b
 ...
 ```
 
 ## Cause
 
-Explain why a user might see this error.
+Purescript requires that the type declaration for a function or value (i.e. the line with the `::` in it)
+be followed immidiatly by the definition of that function or value (i.e. the line with the `=` in it). The
+compiler has found a declaration for which a defination did not follow.
```
Comment (JordanMartinez) [inline errors/OrphanTypeDeclaration.md:17]:
```diff
- be followed immidiatly
+ be followed immediately
```
```diff
- for which a defination
+ for which a definition
```

File: errors/OrphanTypeDeclaration.md [hunk 2]
```diff
@@ -5,17 +5,23 @@
...
 
-## Fix
-
-- Suggest possible solutions.
+The most common cause of this error are typographical issues.  Check to see that the declaration and the
```
Comment (JordanMartinez) [inline errors/OrphanTypeDeclaration.md:19]: How about "This is often caused by typographical issues." ?

File: errors/OrphanTypeDeclaration.md [hunk 3]
```diff
@@ -5,17 +5,23 @@
...
+definition are spelled the same. Purescript is a whitespace sensitive language and the presence of an
+extra space can make the defination look like a part of the declaration.
```
Comment (JordanMartinez) [inline errors/OrphanTypeDeclaration.md:21]: `defination` -> `definition`

File: errors/OrphanTypeDeclaration.md [hunk 4]
```diff
@@ -5,17 +5,23 @@
...
 
-## Notes
+## Fix
 
-- Additional notes.
+- Ensure that the next (non-comment) line is the definition for the listed declaration.
+- Check that the spelling of they two matches.
```
Comment (JordanMartinez) [inline errors/OrphanTypeDeclaration.md:26]: `they` -> `the`

File: errors/OrphanTypeDeclaration.md [hunk 5]
```diff
@@ -5,17 +5,23 @@
...
+- Check for extra spaces.  Both declarationa and defination should start with the same indentation.
```
Comment (JordanMartinez) [inline errors/OrphanTypeDeclaration.md:27]: `declarationa` -> `declaration`
`definition` -> `definition`

**Review by JordanMartinez — CHANGES_REQUESTED** [review]
Thanks for the clear write up. Could you fix some of the spelling issues here?

**Comment by antarestrader** [comment]
Sorry about the typos.  Perhaps that is why I see this error so often.  That will teach me to try to edit directly on GitHub without the benefit of a spell check.

**Review by JordanMartinez — APPROVED** [review]
Thanks!

File: errors/OrphanTypeDeclaration.md [hunk 6]
```diff
@@ -5,17 +5,23 @@
...
 
-Explain why a user might see this error.
+Purescript requires that the type declaration for a function or value (i.e. the line with the `::` in it)
+be followed immediately by the definition of that function or value (i.e. the line with the `=` in it). The
+compiler has found a declaration for which a definition did not follow.
 
-## Fix
-
-- Suggest possible solutions.
+This is often caused by typographical issues.  Check to see that the declaration and the
```
Comment (hdgarrood) [inline errors/OrphanTypeDeclaration.md:19]: Just a suggestion; while “typographical issues” is correct here, I think there’s a risk of not getting the point across (I suspect many people don’t know that this is what “typo” is short for).
```suggestion
This is often caused by spelling mistakes.  Check to see that the declaration and the
```

File: errors/OrphanTypeDeclaration.md [hunk 7]
```diff
@@ -5,17 +5,23 @@
...
+definition are spelled the same. Purescript is a whitespace sensitive language and the presence of an
```
Comment (hdgarrood) [inline errors/OrphanTypeDeclaration.md:20]:
```suggestion
definition are spelled the same. PureScript is a whitespace sensitive language and the presence of an
```

File: errors/OrphanTypeDeclaration.md (hunk 6 above)
Comment (JordanMartinez) [reply errors/OrphanTypeDeclaration.md:19]: I almost said this in my review...

Comment (antarestrader) [reply errors/OrphanTypeDeclaration.md:19]: I thought about this.  This error also shows up with indentation issues which are not spelling errors technically speaking, but I have no problem if the community wants to use simpler language.

Comment (milesfrain) [reply errors/OrphanTypeDeclaration.md:19]: I missed the error myself when reading the above example.
What about:
> This is often caused by spelling mistakes ("Two" vs "Too" in this case).

**Pull request merged**